src/sentry/static/sentry/app/components/alertLink.tsx

```diff
@@ -1,29 +1,34 @@
 import styled from 'react-emotion';
-import PropTypes from 'prop-types';
 import React from 'react';
+import {LocationDescriptor} from 'history';
 
 import Link from 'app/components/links/link';
 import InlineSvg from 'app/components/inlineSvg';
 import space from 'app/styles/space';
 
-export default class AlertLink extends React.Component {
-  static propTypes = {
-    to: PropTypes.string,
-    href: PropTypes.string,
-    icon: PropTypes.string,
-    priority: PropTypes.oneOf(['info', 'warning', 'success', 'error', 'muted']),
-    size: PropTypes.oneOf(['small', 'normal']),
-  };
+type Size = 'small' | 'normal';
+type Priority = 'info' | 'warning' | 'success' | 'error' | 'muted';
+
+type Props = {
+  // do we need to enforce here to always provide either href or to?
+  href?: string;
+  to?: LocationDescriptor;
+  icon?: string;
+  size: Size;
+  priority: Priority;
+  onClick?: (e: React.MouseEvent) => void;
+};
```

Comment:
The keys cannot have any logic, so you'll need to type it like this:

```
type PropsWithHref = {
  href: string;
}
type PropsWithTo = {
  to: LocationDescriptor;
}
type Props = (PropsWithTo | PropsWithHref) & {
  icon?: string;
  size: Size;
  priority: Priority;
  onClick?: (e: React.MouseEvent) => void;
}
```

Reply:
thanks, I updated the PR